flink-ml-lib/src/test/java/org/apache/flink/ml/feature/KBinsDiscretizerTest.java

```diff
@@ -0,0 +1,296 @@
+                    Row.of(Vectors.dense(7.25, 3, 4)),
+                    Row.of(Vectors.dense(13, 4, 5)),
+                    Row.of(Vectors.dense(15, 4, 6)));
+
+    private static final double[][] UNIFORM_MODEL_DATA =
+            new double[][] {
+                new double[] {1, 5, 9, 13},
+                new double[] {Double.MIN_VALUE, Double.MAX_VALUE},
+                new double[] {0, 1, 2, 3}
+            };
+
+    private static final List<Row> UNIFORM_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 1)),
+                    Row.of(Vectors.dense(0, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)));
+
+    private static final List<Row> QUANTILE_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(1, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)));
+
+    private static final List<Row> KMEANS_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 1)),
+                    Row.of(Vectors.dense(0, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)));
+
+    private static final double TOLERANCE = 1e-7;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        trainTable = tEnv.fromDataStream(env.fromCollection(TRAIN_INPUT)).as("input");
+        testTable = tEnv.fromDataStream(env.fromCollection(TEST_INPUT)).as("input");
+    }
+
+    @SuppressWarnings("all")
+    private void verifyPredictionResult(
+            List<Row> expectedOutput, Table output, String predictionCol) throws Exception {
+        List<Row> collectedResult =
+                IteratorUtils.toList(
+                        tEnv.toDataStream(output.select($(predictionCol))).executeAndCollect());
+        compareResultCollections(
+                expectedOutput,
+                collectedResult,
+                (o1, o2) -> {
```

Comment:
There is a method called `TestUtils::compare`, which might be useful here. And #142 has been generalizing the method from `DenseVector` to `Vector`.